llvm/include/llvm/TargetParser/AArch64TargetParser.h

```diff
@@ -632,7 +632,18 @@ inline constexpr CpuInfo CpuInfos[] = {
                                AArch64::AEK_PAUTH, AArch64::AEK_SVE2BITPERM,
                                AArch64::AEK_FLAGM, AArch64::AEK_PERFMON,
                                AArch64::AEK_PREDRES, AArch64::AEK_PROFILE})},
-    {"cortex-r82", ARMV8R, AArch64::ExtensionBitset({AArch64::AEK_LSE})},
+    {"cortex-r82", ARMV8R,
+     AArch64::ExtensionBitset({AArch64::AEK_CRC, AArch64::AEK_DOTPROD,
```

Comment:
Are these implied by ARMV8R? Is there an advantage to having them specified separately too?

Reply:
Thanks, good spot, yes these can be trimmed.

- AEK_FP is enabled for ARMV8A
- AEK_LSE is enabled for ARMV8_1A
- AEK_CRC is enabled for ARMV8_1A
- AEK_RDM is enabled for ARMV8_1A
- AEK_RAS is enabled for ARMV8_2A
- AEK_RCPC is enabled for ARMV8_3A
- AEK_PAUTH is enabled for ARMV8_3A
- AEK_DOTPROD is enabled for ARMV8_4A
- AEK_FLAGM doesn't default
- AEK_PERFMON doesn't default
- AEK_PREDRES doesn't default

ARMV8R inherits from ARMV8_5A.DefaultExts, so we only need the latter 3 flags.

Code now updated.

Reply:
Can you check if ARMV8R enables AEK_LSE? There is a flip(), which might be turning it off?

Reply:
Ah yes. Now definitely enabled in the latest commit.